docs/src/modules/components/AppFrame.js

```diff
@@ -142,229 +143,211 @@ const styles = theme => ({
   },
 });
 
-class AppFrame extends React.Component {
-  state = {
-    languageMenu: null,
-    mobileOpen: false,
-  };
+function AppFrame(props) {
+  const { children, classes, theme, t, userLanguage } = props;
 
-  componentDidMount() {
-    const { canonical } = pathnameToLanguage(window.location.pathname);
-    this.canonical = canonical;
+  const [languageMenu, setLanguageMenu] = React.useState(null);
+  function handleLanguageIconClick(event) {
+    setLanguageMenu(event.currentTarget);
+  }
+  function handleLanguageMenuClose() {
+    setLanguageMenu(null);
   }
 
-  handleDrawerOpen = () => {
-    this.setState({ mobileOpen: true });
-  };
-
-  handleDrawerClose = () => {
-    this.setState({ mobileOpen: false });
-  };
-
-  handleLanguageIconClick = event => {
-    this.setState({ languageMenu: event.currentTarget });
-  };
-
-  handleLanguageMenuClose = () => {
-    this.setState({ languageMenu: null });
-  };
+  const [mobileOpen, setMobileOpen] = React.useState(false);
+  function handleDrawerOpen() {
+    setMobileOpen(true);
+  }
+  function handleDrawerClose() {
+    setMobileOpen(false);
+  }
 
-  handleTogglePaletteType = () => {
-    const paletteType = this.props.reduxTheme.paletteType === 'light' ? 'dark' : 'light';
+  const changeTheme = useChangeTheme();
+  function handleTogglePaletteType() {
+    const paletteType = theme.palette.type === 'light' ? 'dark' : 'light';
     document.cookie = `paletteType=${paletteType};path=/;max-age=31536000`;
 
-    this.props.dispatch({
-      type: ACTION_TYPES.THEME_CHANGE,
-      payload: {
-        paletteType,
-      },
-    });
-  };
+    changeTheme({ paletteType });
+  }
+  function handleToggleDirection() {
+    changeTheme({ direction: theme.direction === 'ltr' ? 'rtl' : 'ltr' });
+  }
 
-  handleToggleDirection = () => {
-    this.props.dispatch({
-      type: ACTION_TYPES.THEME_CHANGE,
-      payload: {
-        direction: this.props.reduxTheme.direction === 'ltr' ? 'rtl' : 'ltr',
-      },
-    });
-  };
+  const canonicalRef = React.useRef();
+  React.useEffect(() => {
+    const { canonical } = pathnameToLanguage(window.location.pathname);
+    canonicalRef.current = canonical;
+  }, []);
 
-  render() {
-    const { children, classes, reduxTheme, t, userLanguage } = this.props;
-    const { languageMenu } = this.state;
+  const { direction } = useTheme();
```

Comment:
Are we using `withStyles({}, { withTheme: true })` and `useTheme()` at the same time? Should we use a single approach?

Reply:
Didn't realized we injected it here. In any case using a single source is likely better.